### PR TITLE
[PLAT-1838] -- Convert IOT demo to Vertex Viewer 2.0

### PR DESCRIPTION
**Pull request by gbiv**
## Summary

Update to the 2.0 viewer.
- Added sync between chart and data table
- Added marker on chart and multiple series (multi dependent scales didn't work )

## Dependencies

[Ticket](https://vertexvis.atlassian.net/browse/PLAT-1838)

## TODO
- [x] Deploy (with Vercel)?

### CODE REVIEW

File: src/components/Home.tsx [hunk 1]
```diff
@@ -1,98 +1,270 @@
...
-          onConfirm={(cs) => {
-            setCredentials(cs);
-            setDialogOpen(false);
+      rightDrawer={
+        <RightDrawer
+          assets={{
+            assets: getAssets(credentials.streamKey),
+            onSelect: async (a: Asset) => {
+              setAsset(a);
+              console.log(a);
```
Comment (amvertex) [inline src/components/Home.tsx:234]: Nit: can remove

File: src/components/RightDrawer.tsx [hunk 2]
```diff
@@ -35,21 +33,56 @@ export function RightDrawer({ files, metadata, open }: Props): JSX.Element {
...
         </AccordionSummary>
-        <RecentFiles files={files} />
+        <Box>
+          <Link
+            href="https://github.com/Vertexvis/iot-demo"
+            rel="noreferrer"
+            sx={{ alignSelf: "center", mr: 2 }}
+            target="_blank"
+          >
+            View on GitHub
```
Comment (amvertex) [inline src/components/RightDrawer.tsx:83]: Nit: You could add a little space on the left before the text starts to align with the data in other sections.

File: src/components/TimeSeriesChart.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  readonly faultCodes: FaultCode[];
+}
+
+export function TimeSeriesChart({
+  sensors,
+  timestamp,
+  faultCodes,
+}: Props): JSX.Element {
+  const colors = useOrdinalColorScale({ scheme: "nivo" }, "id");
+  // const sensors =[sensor]
```
Comment (amvertex) [inline src/components/TimeSeriesChart.tsx:20]: Can delete

File: src/components/TimeSlider.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    setMinTimestamp(sensor.data[0].timestamp);
+    setMaxTimestamp(sensor.data[sensor.data.length - 1].timestamp);
+  }, [sensor]);
+  React.useEffect(() => {
+    if (sensor === undefined) {
+      return;
+    }
+    const step = sensor.data.findIndex((x) => x.timestamp === timestamp);
+    setActiveStep(step);
+    // }
```
Comment (amvertex) [inline src/components/TimeSlider.tsx:39]: Can delete

File: src/lib/scene-items.ts [hunk 5]
```diff
@@ -1,46 +1,145 @@
...
+
+  const scene = await viewer.scene();
+  if (scene == null) return;
+
+  await scene
+    .items((op) =>
+      groups.map((g) => {
+        const w = op.where((q) => q.withSuppliedIds(g.suppliedIds));
+        return apply
+          ? w.materialOverride(ColorMaterial.fromHex(g.color, 178))
```
Comment (amvertex) [inline src/lib/scene-items.ts:52]: I would maybe save the opacity value of 178 as a constant and then reference it here and in the next method in order to be clearer what the number is representing.

**Review by amvertex — COMMENTED** [review]
The demo looks great! I tested with the vercel deploy and it worked really well. Great job!

File: src/components/RightDrawer.tsx (hunk 2 above)
Comment (gbiv) [reply src/components/RightDrawer.tsx]: Just moved this to a button on the speed dial toolbar

Comment (amvertex) [reply src/components/RightDrawer.tsx]: I like it there!

**Review by amvertex — APPROVED** [review]
Great job!

**Pull request merged**